image_transport/include/image_transport/image_transport.h

```diff
@@ -35,149 +35,191 @@
+    rclcpp::Node::SharedPtr node,
+    const std::string & base_topic,
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default);
+
+/**
+ * \brief Subscribe to an image topic, free function version.
+ */
+Subscriber create_subscription(
+    rclcpp::Node::SharedPtr node,
+    const std::string & base_topic,
+    const Subscriber::Callback& callback,
+    const std::string& transport,
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default);
+
+/*!
+ * \brief Advertise a camera, free function version.
+ */
+CameraPublisher create_camera_publisher(
+    rclcpp::Node::SharedPtr node,
+    const std::string & base_topic,
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default);
+
+/*!
+ * \brief Subscribe to a camera, free function version.
+ */
+CameraSubscriber create_camera_subscription(
+    rclcpp::Node::SharedPtr node,
+    const std::string & base_topic,
+    const CameraSubscriber::Callback & callback,
+    const std::string& transport,
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default);
+
+std::vector<std::string> getDeclaredTransports();
+std::vector<std::string> getLoadableTransports();
+
 /**
  * \brief Advertise and subscribe to image topics.
  *
  * ImageTransport is analogous to ros::NodeHandle in that it contains advertise() and
  * subscribe() functions for creating advertisements and subscriptions of image topics.
- */
+*/
 class ImageTransport
 {
 public:
-  explicit ImageTransport(const ros::NodeHandle& nh);
+  explicit ImageTransport(rclcpp::Node::SharedPtr node);
 
   ~ImageTransport();
 
   /*!
    * \brief Advertise an image topic, simple version.
    */
-  Publisher advertise(const std::string& base_topic, uint32_t queue_size, bool latch = false);
-
-  /*!
-   * \brief Advertise an image topic with subcriber status callbacks.
-   */
-  Publisher advertise(const std::string& base_topic, uint32_t queue_size,
-                      const SubscriberStatusCallback& connect_cb,
-                      const SubscriberStatusCallback& disconnect_cb = SubscriberStatusCallback(),
-                      const ros::VoidPtr& tracked_object = ros::VoidPtr(), bool latch = false);
+  Publisher advertise(
+    const std::string & base_topic,
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default);
 
   /**
-   * \brief Subscribe to an image topic, version for arbitrary boost::function object.
+   * \brief Subscribe to an image topic, version for arbitrary std::function object.
    */
-  Subscriber subscribe(const std::string& base_topic, uint32_t queue_size,
-                       const boost::function<void(const sensor_msgs::ImageConstPtr&)>& callback,
-                       const ros::VoidPtr& tracked_object = ros::VoidPtr(),
-                       const TransportHints& transport_hints = TransportHints());
+  Subscriber subscribe(
+    const std::string & base_topic,
+    const Subscriber::Callback& callback,
+    const std::string& transport,
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default);
```

Comment:
It would be helpful for porting existing code if *at least additionally) the previous function signature is maintained. So in this case keep a `queue_size` parameter and set it internally on the profile.

Same applies to other signatures.

Reply:
I guess there are several other API changes which will make it a not straight forward conversion at this point.

It looks like the hints have been removed from several signature but instead the transport string has been introduced. What is the plan around the `TransportHints`?

Reply:
Can do for `queue_size`.
I had originally cut `TransportHints` because it would require a parameters_client internally in order to retrieve the string parameter.  The only other thing it was doing was interacting with the ROS1 transport hints, so it seemed easier to pass a string in from the outside.

I can make another similar structure if keeping the API the same is important.

Reply:
I have ported the `web_video_server` package already. I just noticed the effort during that work. I am not sure if anything should be done but every change in the API (as small as it might be) will be perceived as an (unnecessary?) burden by every user / developer when porting their code.

Reply:
Sounds reasonable, can I do it in a second PR so I can get this one merged?

Reply:
Absolutely. Not a blocker.